src/components/SearchScreen/SearchScreen.bs

```diff
@@ -45,41 +44,45 @@ function OnTextChange() as void
     end if
     text = m.keyboard.text
 
-    createTaskPromise("SearchSuggestionsTask", {
-        input: {
-            q: text
-        }
-    }).then(function(task as object) as void
-        ' In case this is an old request, discard suggestions
-        if task.output.q <> m.keyboard.text
-            return
+    m.searchSuggestionsTask = CreateObject("roSGNode", "SearchSuggestionsTask")
+    m.searchSuggestionsTask.id = "SearchSuggestionsTask"
+    m.searchSuggestionsTask.input = { q: text }
+    m.searchSuggestionsTask.ObserveField("output", "OnSearchSuggestionsTaskResults")
+    m.searchSuggestionsTask.control = "RUN"
+end function
+
+function OnSearchSuggestionsTaskResults() as void
+    task = m.searchSuggestionsTask
+    ' In case this is an old request, discard suggestions
+    if task.output.q <> m.keyboard.text
+        return
+    end if
```

Comment:
TODO: this needs to be fixed - by cancelling running task before starting a new one